### PR TITLE
Http proxy

### PR DESCRIPTION
**Pull request by claudiaop**
Added changes after Costin's review

### CODE REVIEW

File: service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+ write "GET /get HTTP/1.1\r\n"
+ write "Host: localhost:8110\r\n"
+ write "User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/40.0.2214.94 Safari/537.36\r\n"
+ write "Connection: keep-alive\r\n"
+ write "Upgrade-Insecure-Requests: 1\r\n"
+ write "\r\n"
+
+ read status "200" /.*/
+
+ write method "DELETE"
```
Comment (Anisotrop) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt]: Why use method "DELETE" if writing again below ?

File: service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+ read /Date: .*/ "\r\n"
+ read "Server: Apache-Coyote/1.1\r\n"
+ read "Transfer-Encoding: chunked\r\n"
+ read "\r\n"
+ read "14\r\n"
+ read "0\r\n"
+ read "\r\n"
+
+
+ write method "DELETE"
```
Comment (Anisotrop) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt]: Why use method "DELETE" if writing again below ?

File: service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+ read /Via: 1.1 kaazing-.+/ "\r\n"
+ read "User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/40.0.2214.94 Safari/537.36\r\n"
+ read "Upgrade-Insecure-Requests: 1\r\n"
+ read "Host: localhost:8080\r\n"
+ read "Connection: close\r\n"
+ read "\r\n"
+
+ write status "200" "OK"
+
+ read method "DELETE"
```
Comment (Anisotrop) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt]: Why use method "DELETE" if reading again below ?

File: service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+ read "DELETE /examples/servlets/servlet/CookieExample HTTP/1.1\r\n"
+ read "User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/40.0.2214.94 Safari/537.36\r\n"
+ read "Host: localhost:8110\r\n"
+ read "Accept-Language: en-US,en\r\n"
+ read "Connection: keep-alive"
+
+ write status "200" "OK"
+
+ read method "DELETE"
```
Comment (Anisotrop) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt]: Why use method "DELETE" if reading again below ?

File: service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.delete.method.rpt [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+ read /Via: 1.1 kaazing-.+/ "\r\n"
+ read "User-Agent: Mozilla/5.0\r\n"
+ read "Upgrade-Insecure-Requests: 1\r\n"
+ read "Host: localhost:8080\r\n"
+ read "Connection: close\r\n"
+ read "\r\n"
+
+ write status "200" "OK"
+
+ read method "DELETE"
```
Comment (Anisotrop) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.delete.method.rpt]: Why use method "DELETE" if reading again below ?

File: service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.delete.method.rpt [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+ read "DELETE /examples/servlets/servlet/CookieExample HTTP/1.1\r\n"
+ read "User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/40.0.2214.94 Safari/537.36\r\n"
+ read "Host: localhost:8110\r\n"
+ read "Accept-Language: en-US,en\r\n"
+ read "Connection: keep-alive"
+
+ write status "200" "OK"
+
+# gateway(sender) closes the connection
+closed
```
Comment (Anisotrop) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.delete.method.rpt]: nit: please add a newline

Comment (claudiaop) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.delete.method.rpt]: added

File: service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.delete.method.rpt (hunk 5 above)
Comment (claudiaop) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.delete.method.rpt]: removed

File: service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt (hunk 4 above)
Comment (claudiaop) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt]: removed

File: service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt (hunk 3 above)
Comment (claudiaop) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt]: removed

File: service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt (hunk 2 above)
Comment (claudiaop) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt]: removed

File: service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt (hunk 1 above)
Comment (claudiaop) [inline service/http.proxy/src/test/scripts/org/kaazing/gateway/service/http/proxy/http.proxy.Idempotent.delete.rpt]: removed

**Pull request merged**